src/casync-tool.c

```diff
@@ -1629,6 +1662,14 @@ static int pull(int argc, char *argv[]) {
                 return r;
         }
 
+        if (arg_rate_limit > 0) {
+                r = ca_remote_set_rate_limit(rr, arg_rate_limit);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to set rate limit: %s\n", strerror(-r));
+                        return r;
+                }
+        }
```

Comment:
All agreed on, except this one. Note that the rate limit is set on both the `CaSync` and the `CaRemote` objects, so we would need to helper functions, which will result in more code, not less.